### PR TITLE
Fix templates for if variables so that they have not to been set

### PR DESCRIPTION
**Pull request by anikaweinmann**
There where problems with templates which include `if`. Here the variables in the `if` part had to be set. This change should fix this, so that the following template can be used without setting `region_union`:

```
{
	"id": "pc_dummy",
	"description": "Placeholder json file",
	"template": {
		"version": "1",
		"list": [
			{
				"module": "r.mapcalc",
				"id": "r.mapcalc_test",
				"inputs": [
					{% if region_union is defined %}
					{
						"param": "region",
						"value": "{{ region_union }}"
					},
					{% endif %}
					{
						"param": "expression",
						"comment": "output = r.mapcalc result, string; value = raster value (default=0.428), float",
						"value": "{{ output }} = {{ value|default(0.428) }}"
					}
			  ]
			}
		]
	}
}

```

### CODE REVIEW

**Comment by anikaweinmann** [comment]
Tests are fixed and linting is done in https://github.com/actinia-org/actinia-module-plugin/pull/37

**Review by mmacata — COMMENTED** [review]
Thanks :)
I added some improvement ideas. If they don't apply, fine for me to be merged like this as well.

File: actinia_module_plugin/core/processing.py [hunk 1]
```diff
@@ -105,8 +106,27 @@ def check_for_errors(undef, parsed_content, kwargs):
         if 'default' in filtered_var[1]:
             default_vars.append(filtered_var[0])
 
+    # find variables which are only in an if statement and has not to be set
+    not_needed_vars = default_vars
+    if "{% if" in tpl_source:
+        for i in undef:
+            if i not in not_needed_vars:
+                r_str = rf"{{% if {i} is defined %}}[\S\n\t\v ]+{{% endif %}}"
```
[the remark this answers is not in the document]
Comment (anikaweinmann) [reply actinia_module_plugin/core/processing.py:114]: I think if the variable is set which should be the case in your examples. The check is not needed. But I can try if your examples will work in a template

File: actinia_module_plugin/core/processing.py [hunk 2]
```diff
@@ -105,8 +106,27 @@ def check_for_errors(undef, parsed_content, kwargs):
...
+                    tpl_source_mod = tpl_source
+                    for if_part in regex_m:
+                        tpl_source_mod = tpl_source_mod.replace(if_part, "")
+                    # check if the variable is still a variable in modified
+                    # template
+                    undef_mod = get_template_undef(tpl_source_mod)
+                    for var in undef:
+                        if var not in not_needed_vars and var not in undef_mod:
+                            not_needed_vars.append(var)
+
```
[the remark this answers is not in the document]
Comment (anikaweinmann) [reply actinia_module_plugin/core/processing.py:126]: Yes only line 117 has to be changed to `append`. I will try it tomorrow.

**Pull request merged**